nncf/common/accuracy_aware_training/runner.py

```diff
@@ -235,11 +236,12 @@ def __init__(self, accuracy_aware_params: Dict[str, object], verbose=True,
         self.cumulative_epoch_count = 0
         self.best_val_metric_value = 0
 
-    def initialize_training_loop_fns(self, train_epoch_fn, validate_fn, configure_optimizers_fn,
+    def initialize_training_loop_fns(self, train_epoch_fn, validate_fn, configure_optimizers_fn, dump_checkpoint_fn,
```

Comment:
Could you provide a signature for `dump_checkpoint_fn`? I think the signature should be common for PyTorch and TensorFlow backends. For example, `dump_checkpoint_fn(model, compression_controller, accuracy_aware_runner, save_dir)` 

Don't forget to update `Usage.md` with your changes.

Reply:
I am suggesting this signature - `dumping_fn(self, model, save_dir, accuracy_aware_metainfo)`, where `accuracy_aware_metainfo` is all accuracy_aware metainfo including compression_controller state

Reply:
If I right understand, you add to a new entity to the public API. I mean `accuracy_aware_metainfo` which AccuracyAwareTrainingRunner provides. At least, we should document keys in the accuracy_aware_metainfo dict.

Could you add test for TensorFlow AA Impl to save checkpoint?

Reply:
> @kshpv Could you please describe the reason for these changes?

Was added